### PR TITLE
update backends tutorials intro

### PR DESCRIPTION
**Pull request by JessicaS11**
Navigating Xarray backends (`engine="rioxarray"`) when using `xr.open_dataset()` can be confusing. One way to help clarify is by guiding the user through determining if the engine they need exists and is built-in, available from another library, or needs to be created. Here we introduce a flowchart to help guide that process.

### CODE REVIEW

**Comment by JessicaS11** [comment]
@TomNicholas @scottyhq @negin513 @betolink Since this was ideated (thanks, Tom!) at our SciPy2024 tutorial planning meeting - any thoughts on where this diagram lives (docs versus tutorial, with a link from the other)?

**Comment by JessicaS11** [comment]
Can anyone help with the configuration options to get this building? To use the `sphinxcontrib.mermaid` extension, `myst-parser` is needed, but as soon as I added that I started getting an ".md extension already registered" error that I can't figure out how to resolve.

**Comment by scottyhq** [comment]
> Can anyone help with the configuration options to get this building? To use the sphinxcontrib.mermaid extension, 

I'm hoping https://github.com/xarray-contrib/xarray-tutorial/pull/270 (bumping sphinx versions) will just remedy this! I'll try to dig into it a bit more soon

**Comment by JessicaS11** [comment]
> I'm hoping https://github.com/xarray-contrib/xarray-tutorial/pull/270 (bumping sphinx versions) will just remedy this! I'll try to dig into it a bit more soon

After some more thought and discussion with others, I'm going to move the diagram and some of this intro content to the [docs](https://docs.xarray.dev/en/stable/user-guide/io.html) as suggested by @dcherian. Then I'll revamp this PR to have a shorter intro that links to those pages.

**Comment by JessicaS11** [comment]
See also (which will need to be merged before this PR): https://github.com/pydata/xarray/pull/9175

File: advanced/backends/backends.md [hunk 1]
```diff
@@ -1,44 +1,48 @@
...
+You can [read different types of files](https://docs.xarray.dev/en/stable/user-guide/io.html) in `xr.open_dataset` by specifying the engine to be used:
 
 ```python
 import xarray as xr
 xr.open_dataset("my_file.grib" , engine="cfgrib")
 ```
 
-For each available engine there is an underlying backend, that reads the data and pack them in a dataset.
+Navigating Xarray backends can be confusing,
+so we recommend checking out [this flow chart](https://docs.xarray.dev/en/stable/user-guide/io.html)
```
Comment (JessicaS11) [inline advanced/backends/backends.md:13]: note to update this link once the diagram is in

File: advanced/backends/backends.md [hunk 2]
```diff
@@ -1,44 +1,48 @@
...
+```
 
-See the [documentation](https://docs.xarray.dev/en/stable/internals/how-to-add-new-backend.html) for more.
+### Links to internal backends
 
-Follow the tutorial on creating a new backend for binary files.
+- [netcdf4](https://pypi.org/project/netCDF4/) - netCDF4
+- [scipy](https://scipy.org/) - netCDF3
+- [zarr](https://pypi.org/project/zarr/) - Zarr
+- [pydap](https://pydap.github.io/pydap/) - Data Access Protocol (DAP/DODS/OPeNDAP)
```
Comment (JessicaS11) [inline advanced/backends/backends.md:40]:
```suggestion
- [pydap](https://pydap.github.io/pydap/) - Data Access Protocol (DAP/DODS/OPeNDAP)
- [h5netcdf](https://h5netcdf.org/) - hdf5
```

**Comment by JessicaS11** [comment]
@scottyhq @dcherian I think this is ready for review... I'd love confirmation that the list of built-in engines is right, because we've got different info in different places.

**Review by scottyhq — APPROVED** [review]
Thanks for this! I think the content flows better top to bottom and the flowchart in the main Xarray docs is great.

**Pull request merged**